ts/screens/profile/ShareDataScreen.tsx

```diff
@@ -1,121 +1,138 @@
 import { IOStyles } from "../../components/core/variables/IOStyles";
-import { RNavScreenWithLargeHeader } from "../../components/ui/RNavScreenWithLargeHeader";
 import I18n from "../../i18n";
 import { setMixpanelEnabled } from "../../store/actions/mixpanel";
-import { isMixpanelEnabled } from "../../store/reducers/persistedPreferences";
-import { GlobalState } from "../../store/reducers/types";
+import { isMixpanelEnabled as isMixpanelEnabledSelector } from "../../store/reducers/persistedPreferences";
 import { getFlowType } from "../../utils/analytics";
 import { useOnFirstRender } from "../../utils/hooks/useOnFirstRender";
-import { useIOStore } from "../../store/hooks";
+import { useIODispatch, useIOSelector, useIOStore } from "../../store/hooks";
+import { IOScrollViewWithLargeHeader } from "../../components/ui/IOScrollViewWithLargeHeader";
+import { IOScrollViewActions } from "../../components/ui/IOScrollView";
 import { trackMixpanelScreen } from "./analytics";
 import {
+  TrackingInfo,
+  trackMixPanelTrackingInfo,
   trackMixpanelDeclined,
   trackMixpanelSetEnabled
 } from "./analytics/mixpanel/mixpanelAnalytics";
 import { useConfirmOptOutBottomSheet } from "./components/OptOutBottomSheet";
 import { ShareDataComponent } from "./components/ShareDataComponent";
 
-type Props = ReturnType<typeof mapDispatchToProps> &
-  ReturnType<typeof mapStateToProps>;
-
-const ShareDataScreen = (props: Props): React.ReactElement => {
+const ShareDataScreen = () => {
+  const timeoutRef = useRef<number>();
   const store = useIOStore();
+  const dispatch = useIODispatch();
+  const isMixpanelEnabled = useIOSelector(isMixpanelEnabledSelector) ?? true;
+  const toast = useIOToast();
+  const shareButtonRef = useRef<View>(null);
+
+  const showUpdatedPreferencesToastMessage = useCallback(() => {
+    const message = I18n.t(
+      "profile.main.privacy.shareData.screen.confirmToast"
+    );
+    toast.success(message);
+    // eslint-disable-next-line functional/immutable-data
+    timeoutRef.current = setTimeout(() => {
+      AccessibilityInfo.announceForAccessibilityWithOptions(
+        I18n.t("profile.main.privacy.shareData.screen.confirmToast"),
+        { queue: true }
+      );
+    }, 1000);
+  }, [toast]);
+
   const { present, bottomSheet } = useConfirmOptOutBottomSheet(() => {
     const flow = getFlowType(false, false);
     trackMixpanelDeclined(flow);
     trackMixpanelSetEnabled(false, flow, store.getState()).finally(() => {
-      props.setMixpanelEnabled(false);
+      dispatch(setMixpanelEnabled(false));
     });
-
-    IOToast.success(
-      I18n.t("profile.main.privacy.shareData.screen.confirmToast")
-    );
+    showUpdatedPreferencesToastMessage();
   });
-  const isMixpanelEnabled = props.isMixpanelEnabled ?? true;
+
+  const dontShareDataLabel = I18n.t(
+    "profile.main.privacy.shareData.screen.cta.dontShareData"
+  );
+  const shareDataLabel = I18n.t(
+    "profile.main.privacy.shareData.screen.cta.shareData"
+  );
```

Comment:
In the [JIRA](https://pagopa.atlassian.net/browse/IOPID-1536) task the screenshot show an **outline** button, while in [FIGMA](https://www.figma.com/design/zaqsSaM3iGl7iSaGoywYNK/Sezione-Profilo?node-id=1831-29141&t=BDQqp7lMhcRuxrUO-4) we have a **solid** button for the deactivation too. Is FIGMA the right one, am I right?

Reply:
You're right, the Figma is the right one!